markdown/sgquery_lexer.go

```diff
@@ -0,0 +1,104 @@
+package markdown
+
+import (
+	"github.com/alecthomas/chroma"
+	. "github.com/alecthomas/chroma" // nolint
+	"github.com/alecthomas/chroma/lexers"
+)
+
+// Sourcegraph query lexer.
+var SGQuery = lexers.Register(chroma.MustNewLexer(
+	&Config{
+		Name:            "Sourcegraph Query",
+		Aliases:         []string{"sgquery", "query"},
+		CaseInsensitive: true,
+	},
+	sgqueryRules(),
+))
+
+//nolint
+func sgqueryRules() Rules {
+	return Rules{
```

Comment:
Maybe add operators? `and`, `or`, `not`, `AND`, `OR`, `NOT`. Should be an easy addition.

Reply:
See line 51 for `and` and `or`. Totally missed `not`, so I'll add that. Everything is case insensitive (line 14), so there shouldn't be a need to worry about different casing

Reply:
me be like ![facepalm](https://user-images.githubusercontent.com/888624/67424842-ca9c2200-f58b-11e9-9c6c-400f8cf39716.gif)

Reply:
Nit, on casing: Strictly speaking we don't support mixed casing like `AnD` or `Or`, so if that happened to be a pattern we would highlight it when we shouldn't I think? nbd.

Reply:
Yeah, I think you're right. I'm not sure if it's possible to turn off case sensitivity for a single rule, but I'll take a look

Reply:
Nah, I would punt.